### PR TITLE
Blazor ebook

### PR DESCRIPTION
**Pull request by csharpfritz**
## Summary

Added Web Forms architecture section

### CODE REVIEW

File: docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md [hunk 1]
```diff
@@ -12,7 +12,15 @@ While ASP.NET Web Forms and Blazor have many similar concepts, they also have di
 
 ## ASP.NET Web Forms
 
-TODO
+ASP.NET Web Forms are built with a page-centric architecture where each request for a location in the application is a separate page that ASP.NET replies with.  As pages are requested, the content of the browser are completely replaced with the results of the page requested.
+
+Pages are made up of HTML markup, C# or Visual Basic Code, a code-behind class that provides logic and event-handling capabilities, and controls.  Controls are reusable units of web user-interface that can be programmatically placed and interacted with on a page.  Pages are made up of files that end with .ASPX containing markup, controls, and some code.  The code-behind classes reside in files with the same base name and an .ASPX.CS or .ASPX.VB extension depending on the programming language used for the code-behind file.  Interestingly, the ASPX file contents are interpreted by the web server and re-compiled whenever they change, even while the web server is already running.
```
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md]: I think we're using lowercase when refering to file extensions elsewhere. Let's be consistent. I believe the docs also typically put file names in *italics*, but @scottaddie should confirm.

File: docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md [hunk 2]
```diff
@@ -12,7 +12,15 @@ While ASP.NET Web Forms and Blazor have many similar concepts, they also have di
...
+
+Controls can be built with markup and delivered as a UserControl with similar structure to the Page with a .ASCX extension and a code-behind class that resides in a .ASCX.CS or .ASCX.VB file.  Controls can also be built completely with code, by inheriting from the WebControl or CompositeControl base classes.
```
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md]:
```suggestion
Controls can be built with markup and delivered as a `UserControl` with similar structure to the Page with a .ASCX extension and a code-behind class that resides in a .ASCX.CS or .ASCX.VB file.  Controls can also be built completely with code, by inheriting from the WebControl or CompositeControl base classes.
```

File: docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md [hunk 3]
```diff
@@ -12,7 +12,15 @@ While ASP.NET Web Forms and Blazor have many similar concepts, they also have di
...
+
+Pages also have an extension event life-cycle.  Each page will raise events for the Initialization, Load, PreRender, and Unload events that occur as the ASP.NET runtime executes the code for the page for every visitor's request.  
+
+Controls on a Page typically post-back to the same page that presented the control, and carry along with them a payload from a hidden form field called ViewState.  The ViewState contains information about the state of the controls at the time they were rendered and presented on the page, allowing the ASP.NET runtime to compare and identify changes in the content submitted to the server.
```
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md]: If the intention is to refer to view state here as a concept, then replace "ViewState" with "view state".

If you're referring to a specific type/member, then use code formatting: `ViewState`

File: docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md [hunk 4]
```diff
@@ -24,7 +32,7 @@ Blazor apps consist of one or more root components that are rendered on an HTML
 
 How the user specifies where components should render and how the components are then wired up for user interactions is [hosting model](./hosting-models) specific.
 
-Blazor [components](./components) are .NET classes that represent a reusable piece of UI. Each component maintains its own state and specifies its own rendering logic, which can include rendering other components. Components specify event handlers for specific user interactions to update the component's state.
+Blazor [components](./components) are .NET classes that represent a reusable piece of UI and resize in files with a RAZOR filename extension.  These components are compiled and deployed as a binary unit, and cannot be updated without recompiling the entire application. Each component maintains its own state and specifies its own rendering logic, which can include rendering other components. Components specify event handlers for specific user interactions to update the component's state.
```
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md]:
```suggestion
Blazor [components](./components) are .NET classes that represent a reusable piece of UI and reside in files with a *.razor* filename extension.  These components are compiled and deployed as a binary unit, and cannot be updated without recompiling the entire application. Each component maintains its own state and specifies its own rendering logic, which can include rendering other components. Components specify event handlers for specific user interactions to update the component's state.
```

**Review by danroth27 — CHANGES_REQUESTED** [review]
Looks good! I have a few quick comments, mostly about formatting. Also it looks like this PR contains some unintended commits?

File: docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md (hunk 1 above)
Comment (scottaddie) [reply docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md]: You're correct.

File: docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md (hunk 3 above)
Comment (csharpfritz) [reply docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md]: Updated to reference the field `ViewState`

**Comment by scottaddie** [comment]
Manually copied over the content from the **ASP.NET Web Forms** section.

**Pull request closed without merging**